langchain-core/src/prompts/chat.ts

```diff
@@ -108,16 +114,27 @@ export class MessagesPlaceholder<
     }
     super(fields);
     this.variableName = fields.variableName;
+    this.optional = fields.optional ?? false;
   }
 
   get inputVariables() {
     return [this.variableName];
   }
 
   validateInputOrThrow(
-    input: Array<unknown>,
+    input: Array<unknown> | undefined,
     variableName: Extract<keyof RunInput, string>
   ): input is BaseMessage[] {
+    if (this.optional && !input) {
+      return false;
+    } else if (!input) {
+      const error = new Error(
+        `Error: Field "${variableName}" in prompt uses a MessagesPlaceholder, which expects an array of BaseMessages as an input value. Received: undefined`
```

Comment:
Suppose on the off chance if someone's using vanilla JS slightly better to `JSON.stringify(input)`

Reply:
I just copied from below